src/WinRT.Runtime/Marshalers.cs

```diff
@@ -1092,7 +1092,7 @@ public void Dispose()
 
         public static new unsafe MarshalerArray CreateMarshalerArray(T[] array)
         {
-#if NET
+#if NET && !NET9_0_OR_GREATER

```

Comment:
We probably should add some non blittable array marshaling test in functional tests for when aot is enabled and are running for .NET 9.

And we should at least manually re-enable the below authoring tests we disabled and confirm it works now on .NET 9.  Note vcxproj will need to be changed to build for .NET 9 which is why I am not sure if we want to do that change in the pipeline yet.

https://github.com/microsoft/CsWinRT/blob/b1733e95c6d35b551fc8cf6fe04e2a0c287346dd/src/Tests/AuthoringConsumptionTest/test.cpp#L263

https://github.com/microsoft/CsWinRT/blob/b1733e95c6d35b551fc8cf6fe04e2a0c287346dd/src/Tests/AuthoringConsumptionTest/test.cpp#L351

Reply:
Will address in follow up PR with testing.